.github/workflows/playwright-changed-files.yml

```diff
@@ -0,0 +1,83 @@
+name: Playwright E2E Tests - Changed Files
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+# Avoid duplicate workflows on same branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-playwright-changed-files
+  cancel-in-progress: true
+
+jobs:
+  test:
+    runs-on: ubuntu-latest-8-cores
```

Comment:
Would a default runner that is free work here too, since we only run the changed tests?

Reply:
The majority of the time in the workflow is spent on installing and building Streamlit. The 8-core machine also speeds up this part, and can speed up the full workflow by 2-3 minutes (which is significant for this workflow to provide value). But keep in mind that if there are no added/changed e2e tests or if there are > 5, the workflow will only run a couple of seconds... so the cost would be almost 0 for those runs.

Reply:
ok gotcha, thanks for clarification! I think eventually it might make sense to move the building of the assets to a dedicated step that runs once and can be re-used by all other workflows. That should also save a lot of time then across the board